framework/src/utils/MooseCoordTransform.C

```diff
@@ -0,0 +1,276 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MooseCoordTransform.h"
+#include "InputParameters.h"
+#include "MultiMooseEnum.h"
+#include "MooseEnum.h"
+#include "MooseMesh.h"
+
+#include "libmesh/tensor_value.h"
+#include "libmesh/point.h"
+
+using namespace libMesh;
+
+MooseCoordTransform::MooseCoordTransform() {}
+MooseCoordTransform::~MooseCoordTransform() {}
+
+void
+MooseCoordTransform::setRotation(const Direction up_direction, const Direction rz_z_axis /*=Y*/)
+{
+  Real alpha = 0, beta = 0, gamma = 0;
+  if (rz_z_axis == Z)
+    mooseError("Invalid rz_coord_axis option of 'Z'");
+  else if (rz_z_axis == INVALID)
+    mooseError("Called MooseCoordTransform::setRotation with INVALID 'rz_z_axis'");
+
+  const Direction rz_r_axis = rz_z_axis == X ? Y : X;
+
+  // Don't error immediately for unit testing purposes
+  bool bad_rz_rotation = false;
+
+  if (up_direction == X)
+  {
+    alpha = 90, beta = 0, gamma = 0;
+    if (rz_r_axis == X)
+    {
+      _rz_r_axis = Y;
+      _rz_z_axis = X;
+    }
+    else if (rz_r_axis == Y)
+    {
+      bad_rz_rotation = true;
+      _rz_r_axis = X;
+      _rz_z_axis = Y;
+    }
+#ifndef NDEBUG
+    else
+      mooseAssert(false, "we should never get here");
```

Comment:
In non-debug, you should get
```
else
  (void)0
```

which the compiler will throw away; so - you shouldn't need the `ifndef`

We should really make an

```
mooseUnreachable()
```

macro or somethin